src/passes/OptimizeInstructions.cpp

```diff
@@ -1520,6 +1517,66 @@ struct OptimizeInstructions
         return nullptr;
     }
   }
+
+  BinaryOp inversedRelationalOp(BinaryOp op) {
```

Comment:
How about making this return `BinaryOp::InvalidBinary` when it can't find the reverse? That would be less ambiguous I think.

Reply:
Yes, it probably more convenient

Reply:
Fixed

Reply:
Makes sense